assets/src/edit-story/app/prepublish/guidance/media.js

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { PAGE_HEIGHT, PAGE_WIDTH } from '../../../constants';
+
+const SAFE_ZONE_AREA = PAGE_HEIGHT * PAGE_WIDTH;
+
+const MIN_PIXEL_DENSITY = 2;
+
+const MAX_VIDEO_LENGTH_SECONDS = 60;
+const MAX_VIDEO_WIDTH = 3840;
+const MAX_VIDEO_HEIGHT = 2160;
+const MIN_VIDEO_HEIGHT = 480;
+const MIN_VIDEO_WIDTH = 852;
+
+// todo: get video frames per second
+// MIN_VIDEO_FPS = 24;
+// export function videoElementFps(element) {}
+
+export function mediaElementSizeOnPage(element) {
```

Comment:
I see this is checking if the area is less than the general area of the safe zone, but should this scope to within the safe zone area parameters? For example a video could be greater than half the size of the safe zone area, but be positioned so that it's not actually using up half of the safe zone area.

Reply:
That's a good case. I'll have to subtract the area that's showing outside of the safe zone and compare those numbers instead. I'll also write a test.

Reply:
well, I changed this to calculate the intersecting area of the two rectangles, but this doesn't take rotation angle into account. :( Going down that rabbit hole now...

Reply:
Went with the `getBoundingRect` util to draw a rectangle and calculate the overlapping area.